tests/components/vevor_heater/test_sensor.py

```diff
@@ -0,0 +1,225 @@
+"""Unit tests for Vevor BLE Heater sensors."""
+from typing import Any, Optional
+from unittest.mock import patch
+
+from bleak import BLEDevice
+from vevor_heater_ble.heater import (
+    HeaterError,
+    OperationalMode,
+    OperationalStatus,
+    PowerStatus,
+    VevorDevice,
+    VevorHeaterStatus,
+)
+
+from homeassistant.components.vevor_heater import DOMAIN
+from homeassistant.components.vevor_heater.const import DEFAULT_UPDATE_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+FIXTURE_BLE_DEVICE = BLEDevice(
+    address="AB:CD:EF:01:23:45", name="FooBar", details=None, rssi=-1
+)
+
+
+async def test_sensors_target_level(
+    enable_bluetooth: None, hass: HomeAssistant
+) -> None:
+    """Tests sensors when the heater is set to fixed level heating."""
+    status_to_set = VevorHeaterStatus(power_status=PowerStatus.OFF)
+
+    with patch(
+        "homeassistant.components.bluetooth.async_ble_device_from_address",
```

Comment:
This would be better a a pytest.fixture since you need to patch it for the duration of the test, and it will save a lot of indent

Reply:
I'm having issues with the interactions between multiple fixtures doing this (patch can also be used as a function decorator, bu that does weird things to the hass parameter). I tried untangling it for a while and then gave up - as this part is significant to what the test does, it may be better to keep it explicit anyway.

Reply:
Can you move it into the patch block below to save some indent and make it a bit more readable instead?

Reply:
Done.